diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +121,181 @@ class LineLength : Rule("line-length") {
+
+    private fun fixComment(wrongNode: ASTNode, configuration: LineLengthConfiguration){
+        var node = wrongNode
+        do {
+            val commentText = node.text
+            var indexLastSpace = configuration.lineLength.toInt()
+            while (commentText[indexLastSpace] != ' ') {
+                --indexLastSpace
+                if (indexLastSpace == 0)
+                    return
+            }
+            val nodeText = "//${commentText.substring(indexLastSpace, commentText.length)}"
+            val newNode = LeafPsiElement(EOL_COMMENT, nodeText)
+            node.treeParent.run {
+                addChild(LeafPsiElement(EOL_COMMENT, commentText.substring(0, indexLastSpace)), node)
+                addChild(PsiWhiteSpaceImpl("\n"), node)
+                addChild(newNode, node)
+                removeChild(node)
+            }
+            node = newNode
+        } while (node.text.length > configuration.lineLength)
+    }
+
+    private fun fixCondition(wrongNode: ASTNode, configuration: LineLengthConfiguration,
+                             leftInitOffset: Int = -1, isProperty: Boolean = false) {
+        var leftOffset = if (leftInitOffset < 0){
+            positionByOffset(wrongNode.firstChildNode.startOffset).second
+        } else leftInitOffset
+        val binList = mutableListOf<ASTNode>()
+        if (isProperty)
+            dfsForProperty(wrongNode, binList)
+        else
+            searchBinaryExpression(wrongNode, binList)
+        lateinit var binaryText: String
+        binList.forEachIndexed { index, astNode ->
+            if (index == 0){
+                binaryText = astNode.text
+            } else {
+                val operationRef = astNode.treeParent.findChildByType(OPERATION_REFERENCE)!!
+                if (operationRef.treePrev.elementType == WHITE_SPACE)
+                    binaryText += operationRef.treePrev.text
+                binaryText += operationRef.text
+                if (operationRef.treeNext.elementType == WHITE_SPACE)
+                    binaryText += operationRef.treeNext.text
+                binaryText+= astNode.text
+                if (leftOffset + binaryText.length > configuration.lineLength) {
+                    val commonParent = astNode.parent({it in binList[index - 1].parents()})!!
+                    val newLine = PsiWhiteSpaceImpl("\n")
+                    commonParent.addChild(newLine, commonParent.findChildByType(OPERATION_REFERENCE)!!.treeNext)
+                    leftOffset = 0
+                    binaryText = astNode.text
+                }
+            }
+        }
+    }
+
+    /**
+     * This method stored all the nodes that have BINARY_EXPRESSION element type.
+     * This method uses recursion to store binary node in the order in which they are located
+     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
+     *@param node node in which to search
+     *@param binList mutable list of ASTNode to store nodes
```

Comment:
Are these nodes really of type `BINARY_EXPRESSION` or do they correspond to parts of the expression?

Reply:
They are both `BINARY_EXPRESSION` and parts of another `BINARY_EXPRESSION`